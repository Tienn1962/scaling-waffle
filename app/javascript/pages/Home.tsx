import React from "react";
import { Link } from "react-router-dom";

export default () => (
  <main class="flex flex-col items-center justify-center h-screen">
    <h1 className="p-4 text-3xl font-bold">Welcome to Scaling Waffle</h1>
    <Link
      to="/recipes"
      className="btn btn-lg"
      role="button"
    >
      View Recipes
    </Link>
  </main>
);
