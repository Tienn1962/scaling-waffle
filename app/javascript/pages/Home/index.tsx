import React from "react";
import { Link } from "react-router-dom";

export default () => (
  <main className="flex flex-col items-center justify-center h-screen">
    <div className="p-4">
      <h1 className="text-3xl font-bold text-center">
        Welcome to the Scaling Waffle
      </h1>
      <h2 className="mt-4 text-1xl text-center">
        The recipes that scale with your hunger
      </h2>
    </div>
    <Link
      to="/recipes"
      className="btn btn-lg"
      role="button"
    >
      View Recipes
    </Link>
  </main>
);
