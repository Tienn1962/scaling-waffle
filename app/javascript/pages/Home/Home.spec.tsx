import * as React from "react";
import { render, screen } from "@testing-library/react";
import Home from ".";
// needed because of the use of <Link /> on the page
import { MemoryRouter } from "react-router-dom";

it("renders the home page", () => {
  const component = render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );

  expect(component).toMatchSnapshot();
});

it("links to the list of recipes", () => {
  const _ = render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );

  expect(screen.getByRole("button").getAttribute("href")).toBe("/recipes");
});
