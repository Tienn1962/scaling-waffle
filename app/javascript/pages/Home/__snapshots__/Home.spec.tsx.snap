// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="flex flex-col items-center justify-center h-screen"
      >
        <div
          class="p-4"
        >
          <h1
            class="text-3xl font-bold text-center"
          >
            Welcome to the Scaling Waffle
          </h1>
          <h2
            class="mt-4 text-1xl text-center"
          >
            The recipes that scale with your hunger
          </h2>
        </div>
        <a
          class="btn btn-lg"
          href="/recipes"
          role="button"
        >
          View Recipes
        </a>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="flex flex-col items-center justify-center h-screen"
    >
      <div
        class="p-4"
      >
        <h1
          class="text-3xl font-bold text-center"
        >
          Welcome to the Scaling Waffle
        </h1>
        <h2
          class="mt-4 text-1xl text-center"
        >
          The recipes that scale with your hunger
        </h2>
      </div>
      <a
        class="btn btn-lg"
        href="/recipes"
        role="button"
      >
        View Recipes
      </a>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
